venues/ICML.cc/2018/RML/webfield/programchairWebfield.js

```diff
@@ -0,0 +1,706 @@
+  var re = AREACHAIR_REGEX;
+  return _.map(
+    _.filter(groups, function(g) { return re.test(g.id); }),
+    function(fg) { return parseInt(fg.id.match(re)[1], 10); }
+  );
+};
+
+var getNotes = function() {
+  return $.getJSON('notes', { invitation: INVITATION, noDetails: true })
+    .then(function(result) {
+      return result.notes;
+    });
+};
+
+var getAllReviews = function(callback) {
+  var invitationId = OFFICIAL_REVIEW_INVITATION;
+  var allNotes = [];
+
+  function getPromise(offset, limit) {
+    return $.getJSON('notes', { invitation: OFFICIAL_REVIEW_INVITATION, offset: offset, limit: limit, noDetails: true })
+    .then(function(result) {
+      allNotes = _.union(allNotes, result.notes);
+      if (result.notes.length == limit) {
+        return getPromise(offset + limit, limit);
+      } else {
+        callback(allNotes);
+      }
+    });
+  };
+
+  getPromise(0, 2000);
+
+};
+
+var getOfficialReviews = function(noteNumbers) {
+
+  var dfd = $.Deferred();
+
+  var noteMap = buildNoteMap(noteNumbers);
+
+  getAllReviews(function(notes) {
+    var re = ANONREVIEWER_REGEX;
+    var ratingExp = /^(\d+): .*/;
+
+    notes.forEach(function(n) {
+      var matches = n.signatures[0].match(re);
+      var num, index, ratingMatch;
+      if (matches) {
+        num = parseInt(matches[1], 10);
+        index = parseInt(matches[2], 10);
+
+        if (num in noteMap) {
+          // Need to parse rating and confidence strings into ints
+          ratingMatch = n.content.rating.match(ratingExp);
+          n.rating = ratingMatch ? parseInt(ratingMatch[1], 10) : null;
+          confidenceMatch = n.content.confidence.match(ratingExp);
+          n.confidence = confidenceMatch ? parseInt(confidenceMatch[1], 10) : null;
+
+          noteMap[num][index] = n;
+        }
+      }
+    });
+    dfd.resolve(noteMap);
+
+  });
+
+  return dfd.promise();
+
+};
+
+var getReviewerGroups = function(noteNumbers) {
+  var noteMap = buildNoteMap(noteNumbers);
+  var reviewerMap = {};
+
+  return $.getJSON('groups', { id: ANONREVIEWER_WILDCARD })
+    .then(function(result) {
+      var re = ANONREVIEWER_REGEX;
+
+      result.groups.forEach(function(g) {
+        var matches = g.id.match(re);
+        var num, index;
+        if (matches) {
+          num = parseInt(matches[1], 10);
+          index = parseInt(matches[2], 10);
+
+          if (g.members.length) {
+            var reviewer = g.members[0];
+            if ((num in noteMap)) {
+              noteMap[num][index] = reviewer;
+            }
+
+            if (!(reviewer in reviewerMap)) {
+              reviewerMap[reviewer] = [];
+            }
+
+            reviewerMap[reviewer].push(num);
+          }
+
+        }
+      });
+      return {
+        byNotes: noteMap,
+        byReviewers: reviewerMap
+      };
+    })
+    .fail(function(error) {
+      displayError();
+      return null;
+    });
+};
+
+var getAreaChairGroups = function(noteNumbers) {
```

Comment:
We can remove this function

Reply:
When I do it gets stuck "loading".  It creates areaChairGroups.byNote that is used elsewhere. When I take out those references too it still gets stuck, but I don't see the error.

Reply:
There is console in the browser to see the errors